sale_purchase_force_vendor/models/stock_move.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_procurement_values(self):
+        res = super()._prepare_procurement_values()
+        if self.sale_line_id.vendor_id:
+            addr = self.sale_line_id.vendor_id.address_get(["invoice"])
```

Comment:
Why are you selecting this address?

Reply:
Reviewing this, better to use the vendor_id field directly.